### ISSUE TITLE
Provisioning redirected user to O365 Developer Program Signup

### ISSUE THREAD

**Issue by gbgljt** [pnp/custom-learning-office-365#107]
#### Category

- [X] Bug

#### Expected or Desired Behavior

Customer tried to install M365 Learning Pathways by following this link:
https://docs.microsoft.com/en-us/office365/customlearning/custom_provision
Expected result was to provision M365 Learning Pathway to their tenant.

#### Observed Behavior

Following the instructions in the document went well until they clicked "Add to your tenant" in step 4.
Instead of being taken to step 5 in the document, the customer was redirected to Office 365 Developer Program Setup.
At this point they abandoned the attempt to set up M365 Learning Pathways and contacted me, as they wanted to understand why this happened and what obligations they will com under if they sign up to the developer program.

#### Steps to Reproduce

The issue was possible to reproduce until the browser window was closed.
Basically they just followed the instructions in the document (although step 2 never appeared again, they came directly to step 3 from step 1).
Today when trying again it was not possible to reproduce. 
Now when following the instructions they still came directly to step 3 from step 1. In step 3 they selected the M365 Learning Pathways, but after clicking "Add to your tenant" in step 4 there was an error message:
![image](https://user-images.githubusercontent.com/48279444/63174718-cfe86800-c042-11e9-8798-1b7675b9caa1.png)

At this point they are stuck and cannot provision the learning pathways.
Customer need help getting around this issue so they can start using the learning pathways.

Thanks for your contribution! Sharing is caring.

**Comment by juliemturner** [comment]
@gbgljt -- Unfortunately, your issue is related to the SharePoint online provisioning service and not Microsoft 365 learning pathways which means we have no expertise to be able to assist you. That said Someone told me this happened to them yesterday but then they tried again and it was fine.  I would suggest they try again to see if it doesn't redirect them to the developer program setup, because it shouldn't. If not, then you will need to submit your issue at https://github.com/SharePoint/sp-dev-docs/issues.

**Comment by gbgljt** [comment]
Hi Julie!

The customer has tried again and again. Only the first day were they re-directed to developer program setup.
The past 3-4 days when they have tried they receive an error after clicking “Add to tenant”:

[cid:image001.jpg@01D5574E.E31816D0]

Expanded error message:
at Microsoft.SharePoint.Client.ClientRequest.ProcessResponseStream(Stream responseStream) at Microsoft.SharePoint.Client.ClientRequest.ProcessResponse() at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryToServerAsync>d__6.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRuntimeContext.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContext.<ExecuteQueryAsync>d__4.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter.GetResult() at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext() in C:\github\PnP-Sites-Core\Core\OfficeDevPnP.Core\Extensions\ClientContextExtensions.cs:line 156 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter.GetResult() at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, Int32 delay, String userAgent) in C:\github\PnP-Sites-Core\Core\OfficeDevPnP.Core\Extensions\ClientContextExtensions.cs:line 91 at SharePointPnP.ProvisioningApp.WebApp.Controllers.HomeController.<LoadThemesFromTenant>d__15.MoveNext() in C:\github\sp-provisioning-service\SharePointPnP.ProvisioningApp\SharePointPnP.ProvisioningApp.WebApp\Controllers\HomeController.cs:line 641 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Runtime.CompilerServices.TaskAwaiter.GetResult() at SharePointPnP.ProvisioningApp.WebApp.Controllers.HomeController.<Provision>d__4.MoveNext() in C:\github\sp-provisioning-service\SharePointPnP.ProvisioningApp\SharePointPnP.ProvisioningApp.WebApp\Controllers\HomeController.cs:line 123 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at System.Web.Mvc.Async.TaskAsyncActionDescriptor.EndExecute(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass8_0.<BeginInvokeAsynchronousActionMethod>b__1(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.CallEndDelegate(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End() at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<InvokeActionMethodFilterAsynchronouslyRecursive>b__11_0() at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_1.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2() at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass7_0.<BeginInvokeActionMethodWithFilters>b__1(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`1.CallEndDelegate(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End() at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethodWithFilters(IAsyncResult asyncResult) at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_6.<BeginInvokeAction>b__3() at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c__DisplayClass3_1.<BeginInvokeAction>b__5(IAsyncResult asyncResult)

Please advise!

/Johan

Johan Tideström
Principal Service Delivery Manager
Microsoft AB

Email/IM: johant@microsoft.com<mailto:johant@microsoft.com>
Phone/Mobile/Teams: +46 8 7522799
Text: +46 734 082799

**Comment by juliemturner** [comment]
@gbgljt -- I'm sorry you're still having issues, but as I said before your issue is related to the SharePoint online provisioning service and not Microsoft 365 learning pathways which means we have no ability or expertise to be able to assist you. To get help you need to submit your issue at https://github.com/SharePoint/sp-dev-docs/issues.

**Comment by gbgljt** [comment]
Hi Julie!

Ok. Sorry for the confusion!
I thought the fact that they were redirected had something to do with the provisioning, but possibly that the error they currently have (“Something went wrong”) possibly had to do with M365 Learning Pathways.

I have now instructed the customer to open a Premier Support case via their admin portal for further help.

Thanks a lot!

/Johan

Johan Tideström
Principal Service Delivery Manager
Microsoft AB

Email/IM: johant@microsoft.com<mailto:johant@microsoft.com>
Phone/Mobile/Teams: +46 8 7522799
Text: +46 734 082799

**Comment by juliemturner** [comment]
@gbgljt  -- nope, until you actually get a site provisioned into the tenant you haven't hit something related to M365 learning pathways.  Unfortunately having your customer open a ticket with Premier Support isn't really going to help them, you just need to open an issue exactly as you did here, but over at the repo I'm pointing you to here https://github.com/SharePoint/sp-dev-docs/issues

**Comment by gbgljt** [comment]
Ok. Again, thanks for the clarification Julie! 😊

/Johan

Johan Tideström
Principal Service Delivery Manager
Microsoft AB

Email/IM: johant@microsoft.com<mailto:johant@microsoft.com>
Phone/Mobile/Teams: +46 8 7522799
Text: +46 734 082799

**Comment by PaoloPia** [comment]
Hi,
The redirection issue should now be fixed. Please test it again and let me know. Thanks.

**Comment by gbgljt** [comment]
Hello!
Well, the redirection issue has not occurred since the first day, almost 2 weeks ago.
As Julie mentioned above I had to open an incident in another forum since provisioning keeps failing.
However, I've learned from other resources that this is because the customer have vanity URLs.
Now trying to do manual install instead.
Thanks!
